crates/wasi-common/cap-std-sync/src/stdio.rs

```diff
@@ -104,6 +108,16 @@ impl WasiFile for Stdin {
     async fn num_ready_bytes(&self) -> Result<u64, Error> {
         Ok(self.0.num_ready_bytes()?)
     }
+    fn isatty(&self) -> bool {
+        #[cfg(unix)]
+        {
+            rustix::io::isatty(&self.0)
+        }
+        #[cfg(windows)]
+        {
+            atty::is(atty::Stream::Stdin)
+        }
```

Comment:
On Linux rustix can use inline asm, while atty always uses libc.

Reply:
My reason for using rustix here was that cap-std-sync already depends on rustix for other things, so this avoids adding a new dependency on `atty` on unix-family platforms.

Rustix's isatty uses the same ioctl as musl, which is a different ioctl from what glibc uses, but the difference shouldn't matter in practice.

Reply:
Hm ok, personally I'm not a fan of having a lot of "if linux else everyone else this" because it seems like it'll just cause tons of Linux-specific or not-Linux-specific issues. This is pretty minor though.

Reply:
The actual change uses `cfg(unix)`, so it isn't Linux-specific.

Reply:
I've now factored this to use a platform-agnostic crate, so that Wasmtime's own code avoids the `cfg`s here.

Reply:
To clarify I'm fine either way, mostly just stating my own personal preference. But also, this looks the same as before, so unsure if this is intended to stay as-is or whether a commit was forgotten? (this PR is fine to merge whenever imo, I don't feel any need to debate fine detatils like this)

Reply:
This one now has `#[cfg(unix)]` and `#[cfg(not(unix))]`. The other is left as `#[cfg(unix)]` and `#[cfg(windows)]` without fallback for non-unix, non-windows platforms.

Reply:
That's consistent with `impl AsFd for File` under `cfg(unix)` which is already there. Ideally we should support non-windows-non-unix, but that's a bigger change.

And, I'm working on a change which will help; I plan to replace all these isatty implementations with a new is-terminal crate which supports windows, unix, and (to the extent possible) non-windows-non-unix with a single API. I'll do it in a separate PR because I need to make a few other changes to avoid cargo deny flagging it for duplicate dependency versions.